generators/client/templates/angular/src/main/webapp/app/admin/configuration/configuration.component.ts.ejs

```diff
@@ -18,46 +18,41 @@
 -%>
 import { Component, OnInit } from '@angular/core';
 
-import { <%=jhiPrefixCapitalized%>ConfigurationService } from './configuration.service';
+import { ConfigurationService, Beans, Bean, PropertySource } from './configuration.service';
 
 @Component({
     selector: '<%= jhiPrefixDashed %>-configuration',
     templateUrl: './configuration.component.html'
 })
-export class <%=jhiPrefixCapitalized%>ConfigurationComponent implements OnInit {
-    allConfiguration: any = null;
-    configuration: any = null;
-    configKeys: any[];
-    filter: string;
-    orderProp: string;
-    reverse: boolean;
+export class ConfigurationComponent implements OnInit {
+    allBeans!: Beans;
+    beans: Bean[] = [];
+    beansFilter = '';
+    beansAscending = true;
+    propertySources: PropertySource[] = [];
 
     constructor(
-        private configurationService: <%=jhiPrefixCapitalized%>ConfigurationService
-    ) {
-        this.configKeys = [];
-        this.filter = '';
-        this.orderProp = 'prefix';
-        this.reverse = false;
-    }
+        private configurationService: ConfigurationService
+    ) {}
+
+    ngOnInit(): void {
+        this.configurationService.getBeans().subscribe(beans => {
+            this.allBeans = beans;
+            this.filterAndSortBeans();
+        });
 
-    keys(dict): string[] {
-        return (dict === undefined) ? [] : Object.keys(dict);
+        this.configurationService.getPropertySources().subscribe(propertySources => (this.propertySources = propertySources));
     }
 
-    ngOnInit() {
-        this.configurationService.get().subscribe((configuration) => {
-            this.configuration = configuration;
+    filterAndSortBeans(): void {
+        this.beans = [];
 
-            for (const config of configuration) {
-                if (config.properties !== undefined) {
-                    this.configKeys.push(Object.keys(config.properties));
-                }
+        Object.keys(this.allBeans).forEach(beanKey => {
```

Comment:
Can we use `filter` here ? looks appropriate

Reply:
Now I'm returning array from service and this allowed write it with filter and very shortly.